core-bundle/src/Resources/public/core.js

```diff
@@ -431,6 +431,127 @@ var AjaxRequest =
+				img = div.getPrevious('div').getElement('img');
+			} else if (div.hasClass('tl_listing_container')) {
+				img = el.getParent('td').getPrevious('td').getFirst('div.list_icon');
+				if (img === null) { // comments
+					img = el.getParent('td').getPrevious('td').getElement('div.cte_type');
+				}
+				if (img === null) { // showColumns
+					img = el.getParent('tr').getFirst('td').getElement('div.list_icon_new');
+				}
+			} else if (next = div.getNext('div')) {
+				if (next.hasClass('cte_type')) {
+					img = next;
+				}
+				if (img === null) { // newsletter recipients
+					img = next.getFirst('div.list_icon');
+				}
+			}
+
+			// Change the row icon
+			if (img !== null) {
+				// Tree view
+				if (img.nodeName.toLowerCase() == 'img') {
+					if (img.getParent('ul.tl_listing').hasClass('tl_tree_xtnd')) {
+						icon = img.get('data-icon');
+						icond = img.get('data-icon-disabled');
+
+						// Prepend the theme path
+						if (icon.indexOf('/') == -1) {
+							icon = AjaxRequest.themePath + (icon.match(/\.svg$/) ? 'icons/' : 'images/') + icon;
+						}
+						if (icond.indexOf('/') == -1) {
+							icond = AjaxRequest.themePath + (icond.match(/\.svg$/) ? 'icons/' : 'images/') + icond;
+						}
+
+						img.src = !published ? icon : icond;
+					} else {
+						pa = img.getParent('a');
+
+						if (pa && pa.href.indexOf('contao/preview') == -1) {
+							if (next = pa.getNext('a')) {
+								img = next.getFirst('img');
+							} else {
+								img = new Element('img'); // no icons used (see #2286)
+							}
+						}
+
+						icon = img.get('data-icon');
+						icond = img.get('data-icon-disabled');
+
+						// Prepend the theme path
+						if (icon.indexOf('/') == -1) {
+							icon = AjaxRequest.themePath + (icon.match(/\.svg$/) ? 'icons/' : 'images/') + icon;
+						}
+						if (icond.indexOf('/') == -1) {
+							icond = AjaxRequest.themePath + (icond.match(/\.svg$/) ? 'icons/' : 'images/') + icond;
+						}
+
+						img.src = !published ? icon : icond;
+					}
+				}
+				// Parent view
+				else if (img.hasClass('cte_type')) {
+					if (!published) {
+						img.addClass('published');
+						img.removeClass('unpublished');
+					} else {
+						img.addClass('unpublished');
+						img.removeClass('published');
+					}
+				}
+				// List view
+				else {
+					icon = img.get('data-icon');
+					icond = img.get('data-icon-disabled');
+
+					// Prepend the theme path
+					if (icon.indexOf('/') == -1) {
+						icon = AjaxRequest.themePath + (icon.match(/\.svg$/) ? 'icons/' : 'images/') + icon;
+					}
+					if (icond.indexOf('/') == -1) {
+						icond = AjaxRequest.themePath + (icond.match(/\.svg$/) ? 'icons/' : 'images/') + icond;
+					}
+
+					img.setStyle('background-image', 'url(' + (!published ? icon : icond) + ')');
+				}
+			}
+		}
+
+		icon = image.get('data-icon') || AjaxRequest.themePath + 'icons/visible.svg';
+		icond = image.get('data-icon-disabled') || AjaxRequest.themePath + 'icons/invisible.svg';
```

Comment:
The ` || AjaxRequest.themePath + 'icons/visible.svg'` part was a BC layer for the `toggleVisibility` method.

https://github.com/contao/contao/blob/4166f4eeccafb9cdcbd15e5e89728bbb4f1f3a81/core-bundle/src/Resources/public/core.js#L453-L456

I would prefer not to have it in the new method, too.

Reply:
Changed in f041bb08e3f284830623cbdde1e3d7da43f02e9b and also removed relative-path handling